core/src/Exceptions/CoreErrorMessages.cs

```diff
@@ -91,5 +91,7 @@ internal class CoreErrorMessages
 
         public const string GetUserNameFailed = "Failed to get user name from the operating system.";
 
+        public const string OAuthNonJsonError = "An error response was returned by the OAuth2 server. Please inspect the exception properties for details.";
```

Comment:
>OAuthNonJsonError [](start = 28, length = 17)

When would we get this specific error? The "NonJson" part seems a little strange.

Reply:
It looks like the server normally sends back a json with error details. We parse the json and give a nice message. But if the server sends a back some other error (and standard HTTP response bodies are not json), then we would have failed with a serialization exception. Now we fail with OAuthNonJsonError. 

Happy to change the name - what do you suggest?

Reply:
UnparseableError? (that's not awesome either but might spark ideas...)

Reply:
Good point...the reason we return this is because the Json is not parseable..."nonjson" isn't clear on that by just looking at it. Or add that to the error message, like you have in the summary for MsalError. "cannot be parsed" something like that.

---
In reply to: [222785065](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1294#discussion_r222785065) [](ancestors = 222785065)